src/App.jsx

```diff
@@ -14,45 +16,122 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABET = 'alphabetical';
+const SORT_BY_LENGTH = 'length';
+
+function getSortedGoodsList(goods, { sortType, isReversed }) {
+  const sortedGoodList = [...goods];
+
+  if (sortType) {
+    sortedGoodList.sort((a, b) => {
+      switch (sortType) {
+        case SORT_BY_LENGTH: {
+          return a.length - b.length;
+        }
+
+        case SORT_BY_ALPHABET: {
+          return a.localeCompare(b);
+        }
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    sortedGoodList.reverse();
+  }
+
+  return sortedGoodList;
+}
+
+export const App = () => {
+  const [sortingOptions, setSortingOptions] = useState({
+    sortType: '',
+    isReversed: false,
+  });
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const visibleGoodsList = getSortedGoodsList(
+    goodsFromServer,
+    sortingOptions,
+  );
+
+  const resetSortingOptions = () => {
+    setSortingOptions({
+      sortType: '',
+      isReversed: false,
+    });
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cl(
+            'button',
+            'is-info',
+            { 'is-light': sortingOptions.sortType !== SORT_BY_ALPHABET },
+          )}
+          onClick={() => setSortingOptions(prevState => ({
+            ...prevState,
+            sortType: SORT_BY_ALPHABET,
+          }))}
```

Comment:
it's better to create own handler function. So you can reuse this handler for reverse button as well

```
const handleSortOptionClick = (option, value) => {
  setSortingOptions(prevState => ({
    ...prevState,
    [option]: value,
  }))
}
```

```suggestion
          onClick={() => handleSortOptionClick('sortType',  SORT_BY_ALPHABET)}
```